demo-server/src/main/java/com/illia/server/request_processor/RequestProcessorImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.illia.server.request_processor;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.illia.server.file_holder.FileHolder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.nio.file.Files;
+
+
+@Service
+public class RequestProcessorImpl implements RequestProcessor {
+
+    @Autowired
+    FileHolder fileHolder;
+
+    @Override
+    public ResponseEntity<Object> proceedDownloadFile(String fileName) {
+        try {
+            var fileHolderResponse = fileHolder.getFile(fileName);
+            if (fileHolderResponse != null) {
+                return ResponseEntity.ok().body(fileHolderResponse);
+            } else {
+                return ResponseEntity.badRequest().body("No such file!".getBytes());
+            }
+        } catch (Exception ex) {
+            return ResponseEntity.internalServerError().build();
+        }
+    }
+
+    // I don't like this realization
```

Comment:
You should not check for returned string to know what happened there, it is not very clean hack